lib/Mount/MountProvider.php

```diff
@@ -300,4 +339,57 @@ private function findConflictsForUser(IUser $user, array $mountPoints): array {
 			return substr($path, 6); // strip leading "files/"
 		}, $paths);
 	}
+
+	/**
+	 * @param list<array{folder_id: int, mount_point: string, permissions: int, quota: int, acl: bool, rootCacheEntry: ?ICacheEntry}> $folders
+	 * @return array<int, list<string>>
+	 */
+	private function findNestedConflicts(int $rootStorageId, array $folders): array {
+		$conflictsByFolder = [];
+
+		// for every folder, create a list of nested folders
+		$mountPoints = [];
+		$pathsToCheck = [];
+		$folderIds = [];
+		foreach ($folders as $folder) {
+			$mountPoint = $folder['mount_point'];
+			$folderId = $folder['folder_id'];
+			$folderIds[$mountPoint] = $folderId;
+
+			// because we sorted the folder by mountpoint, parent folders will always occur before children
+			foreach ($mountPoints as $possibleParent) {
+				if (str_starts_with($mountPoint, $possibleParent)) {
+					$parentFolderId = $folderIds[$possibleParent];
+					$relativeMountPoint = substr($mountPoint, strlen($possibleParent) + 1);
+					$pathsToCheck[] = "__groupfolders/$parentFolderId/$relativeMountPoint";
+				}
+			}
+			$conflictsByFolder[$folderId] = [];
+			$mountPoints[] = $mountPoint;
+		}
+
+		if (!$pathsToCheck) {
+			return [];
+		}
+
+		$pathHashes = array_map('md5', $pathsToCheck);
+
+		$query = $this->connection->getQueryBuilder();
+		$query->select('path')
+			->from('filecache')
+			->where($query->expr()->eq('storage', $query->createNamedParameter($rootStorageId, IQueryBuilder::PARAM_INT)))
+			->andWhere($query->expr()->in('path_hash', $query->createParameter('chunk')));
+
+		$paths = [];
+		foreach (array_chunk($pathHashes, 1000) as $chunk) {
+			$query->setParameter('chunk', $chunk, IQueryBuilder::PARAM_STR_ARRAY);
+			foreach ($query->executeQuery()->fetchAll(\PDO::FETCH_COLUMN) as $conflictPath) {
+				[, $folderId, $relativePath] = explode('/', $conflictPath, 3);
+				$conflictsByFolder[(int)$folderId][] = $relativePath;
+			}
+			$paths = array_merge($paths, );
```

Comment:
this doesn't seem right. otoh $paths isn't read anyway so it doesn't do any harm. but the variable can probably go.

Reply:
cc @icewind1991